pkg/adapter/receive_adapter_test.go

```diff
@@ -156,21 +157,24 @@ var testCases = []testCase{
 func TestGracefulShutdown(t *testing.T) {
 	ce := adaptertest.NewTestClient()
 	ra := newTestAdapter(t, ce)
-	ctx, cancel := context.WithCancel(context.TODO())
-
-	go func() {
-		t.Logf("starting webhook server")
+	ctx, cancel := context.WithCancel(context.Background())
+	done := make(chan struct{})
+	go func(ctx context.Context) {
 		err := ra.Start(ctx)
 		if err != nil {
-			t.Error(err)
+			t.Error("Unexpected error:", err)
 		}
-		cancel()
-	}()
+		close(done)
```

Comment:
It's usually good to `defer` this at the start of the goroutine to guarantee it'll be called.

Reply:
good point!